### PR TITLE
Added add feature & general code clean up

### PR DESCRIPTION
**Pull request by Poopies99**
1. Removed fetchMod method from Duke class
2. Remove SetupTT method from Duke class
3. Added modules data back
4. Completed AddCommand
5. Updated parseCommand to incorporate the semester of the current timetable
6. Updated the run method in Duke after changes from 2.
7. Added getSemesterData method in Module.java
8. Added getSemester method in TimeTable.java
9. Created a new directory to store timetable modules that have been added
10. Added Created lessonDetail method in Lesson class to return print values
11. Added addTimetableFile method to save added modules
12. Added comment to clarify return new Semester in getting Semester method
13. Added new methods to assist for addCommand
14. Added methods for TextUI

### CODE REVIEW

File: src/main/java/seedu/command/AddCommand.java [hunk 1]
```diff
@@ -1,36 +1,110 @@
...
+        if (semesterData.isTutorialExist()) {
+            Lesson tutorial = selectTutorial(semesterData);
+            addedLesson.add(tutorial);
+        }
+        if (semesterData.isLabExist()) {
+            Lesson lab = selectLab(semesterData);
+            addedLesson.add(lab);
+        }
+        Module addedModule = module;
+        addedModule.getSemester(semester).changeTimetable(addedLesson);
```
Comment (adricpjw) [inline src/main/java/seedu/command/AddCommand.java:55]: Would this work  with modules that have multiple lectures or multiple tutorials like CS2113T?

File: src/main/java/seedu/storage/ModStorage.java [hunk 2]
```diff
@@ -115,6 +114,13 @@ public static void showModOffline(String moduleCode) {
         TextUi.printLocalSearchMessage();
     }
 
+    public static void addTimetableFile(Module module) throws IOException {
+        String filePath = "timetable/" + module.getModuleCode() + ".json";
```
Comment (ChewyChair) [inline src/main/java/seedu/storage/ModStorage.java:118]: Should we move this to the data folder (e.g. data/timetable/) ?

File: src/main/java/seedu/ui/TextUi.java [hunk 3]
```diff
@@ -121,4 +122,27 @@ public static void printLocalSearchMessage() {
...
+    }
+
+    public static void printLaboratoryMessage() {
+        System.out.println("Which Laboratory time slot would you like to add ?");
+        System.out.println("Laboratory Time slots:");
+    }
+
+    public static void printLessonInfo(int serial, Lesson lesson) {
+        System.out.println(serial + ": " + lesson.lessonDetail());
+    }
```
Comment (ChewyChair) [inline src/main/java/seedu/ui/TextUi.java:147]: Referring to the 3 json files below and my previous comment, would it be better to move the jsons into /data/ so git does not track them for now?

File: src/main/java/seedu/storage/ModStorage.java [hunk 4]
```diff
@@ -115,6 +114,13 @@ public static void showModOffline(String moduleCode) {
...
+        FileWriter fw = new FileWriter(filePath);
+        Gson gson = new Gson();
+        gson.toJson(module, fw);
+    }
+
```
Comment (adricpjw) [inline src/main/java/seedu/storage/ModStorage.java:123]: Perhaps liaise with @xuanlc113 to work out timetable local save

**Comment by Poopies99** [comment]
Noted, thanks for the feedback will create another PR for this issue

**Pull request closed without merging**